### ISSUE TITLE
ArangoDatabase.explain() throws exception on joined queries

### ISSUE THREAD

**Issue by PeekAndPoke** [arangodb/arangodb-java-driver#314]
Hi there!

**My environment is:**
ArangoDB: 3.5.0
arangodb-java-driver: 6.2.0 

**Bug**
explain() works fine for 'simple' queries. For joined queries like the following the deserialization fails:

```
FOR `character` IN `got_characters`
    FOR `actor` IN `got_actors`
        FILTER `character`.`actor` == `actor`.`_id`
        RETURN `character`
```

So calling 

`arangoDb.explain(query, new HashMap(), new AqlQueryExplainOptions())`

results in the following error:

```
Caused by: com.arangodb.velocypack.exception.VPackParserException: com.arangodb.velocypack.exception.VPackValueTypeException: Expecting type OBJECT
        at com.arangodb.velocypack.VPack.deserialize(VPack.java:413)
        at com.arangodb.internal.util.ArangoDeserializerImpl.deserialize(ArangoDeserializerImpl.java:55)
        ... 80 more
Caused by: com.arangodb.velocypack.exception.VPackValueTypeException: Expecting type OBJECT
        at com.arangodb.velocypack.VPackSlice.objectIterator(VPackSlice.java:772)
        at com.arangodb.velocypack.VPack.deserializeFields(VPack.java:523)
        at com.arangodb.velocypack.VPack.deserializeObject(VPack.java:466)
        at com.arangodb.velocypack.VPack.getValue(VPack.java:603)
        at com.arangodb.velocypack.VPack.deserializeField(VPack.java:539)
        at com.arangodb.velocypack.VPack.deserializeFields(VPack.java:527)
        at com.arangodb.velocypack.VPack.deserializeObject(VPack.java:466)
        at com.arangodb.velocypack.VPack.getValue(VPack.java:603)
        at com.arangodb.velocypack.VPack.deserializeCollection(VPack.java:631)
        at com.arangodb.velocypack.VPack.getValue(VPack.java:581)
        at com.arangodb.velocypack.VPack.deserializeField(VPack.java:539)
        at com.arangodb.velocypack.VPack.deserializeFields(VPack.java:527)
        at com.arangodb.velocypack.VPack.deserializeObject(VPack.java:466)
        at com.arangodb.velocypack.VPack.getValue(VPack.java:603)
        at com.arangodb.velocypack.VPack.deserializeField(VPack.java:539)
        at com.arangodb.velocypack.VPack.deserializeFields(VPack.java:527)
        at com.arangodb.velocypack.VPack.deserializeObject(VPack.java:466)
        at com.arangodb.velocypack.VPack.getValue(VPack.java:603)
        at com.arangodb.velocypack.VPack.deserialize(VPack.java:411)
```

I debugged into it. The response coming from the database looks like this:

```
{
	"plan": {
		"nodes": [{
			"type": "SingletonNode",
			"dependencies": [],
			"id": 1,
			"estimatedCost": 1.0,
			"estimatedNrItems": 1
		}, {
			"type": "EnumerateCollectionNode",
			"dependencies": [1],
			"id": 2,
			"estimatedCost": 45.0,
			"estimatedNrItems": 43,
			"random": false,
			"indexHint": {
				"forced": false,
				"type": "none"
			},
			"outVariable": {
				"id": 0.0,
				"name": "character"
			},
			"projections": [],
			"producesResult": true,
			"database": "kotlindev",
			"collection": "got_characters",
			"satellite": false
		}, {
			"type": "IndexNode",
			"dependencies": [2],
			"id": 7,
			"estimatedCost": 87.785,
			"estimatedNrItems": 43,
			"outVariable": {
				"id": 1.0,
				"name": "actor"
			},
			"projections": [],
			"producesResult": false,
			"database": "kotlindev",
			"collection": "got_actors",
			"satellite": false,
			"needsGatherNodeSort": false,
			"indexCoversProjections": false,
			"indexes": [{
				"id": "0",
				"type": "primary",
				"name": "primary",
				"fields": ["_key"],
				"selectivityEstimate": 1.0,
				"unique": true,
				"sparse": false
			}],
			"condition": {
				"type": "n-ary or",
				"typeID": 63,
				"subNodes": [{
					"type": "n-ary and",
					"typeID": 62,
					"subNodes": [{
						"type": "compare ==",
						"typeID": 25,
						"excludesNull": false,
						"subNodes": [{
							"type": "attribute access",
							"typeID": 35,
							"name": "actor",
							"subNodes": [{
								"type": "reference",
								"typeID": 45,
								"name": "character",
								"id": 0
							}]
						}, {
							"type": "attribute access",
							"typeID": 35,
							"name": "_id",
							"subNodes": [{
								"type": "reference",
								"typeID": 45,
								"name": "actor",
								"id": 1
							}]
						}]
					}]
				}]
			},
			"sorted": true,
			"ascending": true,
			"reverse": false,
			"evalFCalls": true,
			"limit": 0
		}, {
			"type": "ReturnNode",
			"dependencies": [7],
			"id": 6,
			"estimatedCost": 130.785,
			"estimatedNrItems": 43,
			"inVariable": {
				"id": 0.0,
				"name": "character"
			},
			"count": true
		}],
		"rules": ["use-indexes", "remove-filter-covered-by-index", "remove-unnecessary-calculations-2"],
		"collections": [{
			"name": "got_actors",
			"type": "read"
		}, {
			"name": "got_characters",
			"type": "read"
		}],
		"variables": [{
			"id": 3.0,
			"name": "2"
		}, {
			"id": 1.0,
			"name": "actor"
		}, {
			"id": 0.0,
			"name": "character"
		}],
		"estimatedCost": 130.785,
		"estimatedNrItems": 43,
		"initialize": true,
		"isModificationQuery": false
	},
	"cacheable": true,
	"warnings": [],
	"stats": {
		"rulesExecuted": 61,
		"rulesSkipped": 0,
		"plansCreated": 2
	},
	"error": false,
	"code": 200
}
```

So I assume, that the structure of **AqlExecutionExplainEntity** does not fit the response.

**Comment by rashtao** [comment]
fixed in https://github.com/arangodb/arangodb-java-driver/commit/1b66cac5e1a1596767bce6acd69a9dd7ea165bca?w=1

**Comment by PeekAndPoke** [comment]
@rashtao Are you sure that the commit you linked above does address the problem I reported? Maybe you pasted a commit for another issue?

**Comment by rashtao** [comment]
Sorry, I updated my comment with the correct link.

**Comment by rashtao** [comment]
fixed in v6.3.0
